mono/metadata/file-io.c

```diff
@@ -489,8 +489,11 @@ ves_icall_System_IO_MonoIO_FindFirst (MonoString *path,
 	ifh = g_new (IncrementalFind, 1);
 	ifh->find_handle = find_handle;
 	ifh->utf8_path = mono_string_to_utf8_checked (path, &error);
-	if (mono_error_set_pending_exception (&error))
+	if (mono_error_set_pending_exception (&error)) {
+		FindClose (find_handle);
```

Comment:
There's a 

```c
/* FIXME make gc suspendable */
```

on both `FindFirst` and `FindNext` but not on `FindClose`. I guess "gc suspendable" is what you get with the `MONO_ENTER_GC_SAFE` etc macros? I don't know the details to make these functions suspendable. I'm fine with either reverting my change to this file or keep it. The `FIXME` is still there for someone more knowledgeable to fix this up once there's time for it.